specs/waku/v2/waku-store.md

```diff
@@ -41,30 +99,46 @@ message HistoryRPC {
 }
 ```
 
-##### HistoryRPC
+### Index
+`Index` has two parts:
+- `digest`: holds the hash value as a sequence of bytes
+- `receivedTime`: the UNIX time at which the message is received by the waku store node
 
-A node MUST send all History messages (`HistoryQuery`, `HistoryResponse`) wrapped inside a
-`HistoryRPC`.
+### PagingInfo
+`PagingInfo` has three components defined below.
+- `pageSize`: A positive integer indicating the number of waku messages that the client wants to see in her response
+- `cursor`: holds the `Index` of a `WakuMessage`
+- `direction`: It is a boolean value indicating the direction of paging, `true` stands for forward paging and `false` means backward
 
-The `request_id` MUST be a uniquely generated string, the `HistoryResponse` and `HistoryQuery` `request_id` MUST match. The `HistoryResponse` message SHOULD contain any messages found
-whose `topic` can be found in the `HistoryQuery` `topics` array. Any message whose `topic` does not match MUST NOT be included.
 
-##### HistoryQuery
+### HistoryQuery
 
 RPC call to query historical messages.
 
-The `topics` field MUST indicate the list of topics to query.
+- The `topics` field MUST indicate the list of topics to query.
+- `PagingInfo` holds the information required for pagination.
 
-##### HistoryResponse
+### HistoryResponse
 
 RPC call to respond to a HistoryQuery call.
+- The `messages` field MUST contain the messages found, these are [`WakuMessage`] types as defined in the corresponding [specification](./waku-message.md).
+- `PagingInfo`  holds paging information based on which the requester can resume its further history queries. 
+
+### HistoryRPC
+
+A node MUST send all History messages (`HistoryQuery`, `HistoryResponse`) wrapped inside a `HistoryRPC`.
+
+- The `request_id` MUST be a uniquely generated string, the `HistoryResponse` and `HistoryQuery` `request_id` MUST match. 
+The `HistoryResponse` message SHOULD contain any messages found whose `topic` can be found in the `HistoryQuery` `topics` array. Any message whose `topic` does not match MUST NOT be included. Also, the `HistoryResponse` messages SHOULD contain messages that fall into the range specified by the `pagingInfo` field of the `HistoryQuery`.
+
+
+
 
-The `messages` field MUST contain the messages found, these are [`WakuMessage`] types as defined in the corresponding [specification](./waku-message.md).
 
 # Changelog
 
 2.0.0-beta1
-Initial draft version. Released [2020-10-06](https://github.com/vacp2p/specs/commit/75b4c39e7945eb71ad3f9a0a62b99cff5dac42cf)
+Initial draft version. Released [2020-10-21](https://github.com/vacp2p/specs/commit/75b4c39e7945eb71ad3f9a0a62b99cff5dac42cf)
```

Comment:
New version

Reply:
addressed

Reply:
@staheri14 there should still be the old version, don't remove the old one. Just add a new one